.github/workflows/packages-build-wpk.yml

```diff
@@ -0,0 +1,98 @@
+name: Packages - Build WPK packages
+on:
+  workflow_dispatch:
+    inputs:
+      tag:
+        description: 'Tag name of the Docker image to be downloaded (Use developer to set branch name. Use auto to set src/VERSION content)'
+        required: false
+        default: auto
+      linux_branch:
+        description: 'Branch name for compiling the Linux package'
+        required: false
+      windows_package:
+        description: 'Link to download package or package name in S3'
+        required: false
+      macos_package:
+        description: 'Link to download package or package name in S3'
+        required: false
+      revision:
+        description: 'Revision used to naming WPK package'
+        required: false
+        default: 1
+
+  workflow_call:
+
+jobs:
+  WPK-generation:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        os: [linux, windows, macos]
+        arch: [amd64]
+      fail-fast: false
+
+    steps:      
+      - name: Cancel previous runs
+        uses: fkirc/skip-duplicate-actions@master
+        with:
+          cancel_others: 'true'
+          github_token: ${{ secrets.GITHUB_TOKEN }}
+          skip_after_successful_duplicate: 'false'
+
+      - uses: actions/checkout@v4
+
+      - name: Set WPK name, CONTAINER_NAME and TAG
+        run: |
+          VERSION=$(sed 's/v\([0-9]*\.[0-9]*\.[0-9]*\)/\1/' $GITHUB_WORKSPACE/src/VERSION)
+          if [ "${{ github.event.inputs.tag }}" == "auto" ]; then echo "TAG=$VERSION" >> $GITHUB_ENV;
+          elif [ "${{ github.event.inputs.tag }}" == "developer" ]; then echo "TAG=${{ github.ref_name }}" >> $GITHUB_ENV;
+          else echo "TAG=${{ github.event.inputs.tag }}" >> $GITHUB_ENV; fi
+          COMMIT_HASH="$(curl -s https://api.github.com/repos/wazuh/wazuh/commits/${{ github.ref_name }} | grep '"sha"' | head -n 1| cut -d '"' -f 4 | cut -c 1-7)"
+          echo "WPK_NAME=wazuh-wpk_${VERSION}-${{ github.event.inputs.revision }}_${{ matrix.os }}_${{ matrix.arch }}_${COMMIT_HASH}.wpk" >> $GITHUB_ENV
+          if [ "${{ matrix.os }}" == "linux" ]; then echo "CONTAINER_NAME=linux_wpk_builder_x86_64" >> $GITHUB_ENV; else echo "CONTAINER_NAME=common_wpk_builder" >> $GITHUB_ENV; fi
+
```

Comment:
Could we add a **_step_** to verify that a package similar to the one we want to build does not already exist in the **S3** repository? 
In case it does exist, indicate link and do not build the package again.

Reply:
Done